src/script/sign.h

```diff
@@ -53,26 +59,456 @@ class MutableTransactionSignatureCreator : public BaseSignatureCreator {
+                        }
+
+                        // Add to map
+                        witness_scripts.emplace(hash, witnessscript);
+                    } else {
+                        // Make sure that the key is the size of pubkey + 1
+                        if (key.size() != CPubKey::PUBLIC_KEY_SIZE + 1 && key.size() != CPubKey::COMPRESSED_PUBLIC_KEY_SIZE + 1) {
+                            throw std::ios_base::failure("Size of key was not the expected size for the type partial signature pubkey");
+                        }
+                        // Read in the pubkey from key
+                        CPubKey pubkey(key.begin() + 1, key.end());
+                        if (!pubkey.IsFullyValid()) {
+                           throw std::ios_base::failure("Invalid pubkey");
+                        }
+
+                        // Read in the signature from value
+                        std::vector<unsigned char> sig;
+                        sig.resize(value_len);
+                        s >> MakeSpan(sig);
+
+                        // Add to list
+                        input.partial_sigs.emplace(pubkey.GetID(), SigPair(pubkey, sig));
+                    }
+                    break;
+                // BIP 32 HD Keypaths and sighash types
+                case PSBT_BIP32_KEYPATH_SIGHASH:
+                    if (in_globals) {
+                        // Make sure that the key is the size of pubkey + 1
+                        if (key.size() != CPubKey::PUBLIC_KEY_SIZE + 1 && key.size() != CPubKey::COMPRESSED_PUBLIC_KEY_SIZE + 1) {
+                            throw std::ios_base::failure("Size of key was not the expected size for the type BIP32 keypath");
+                        }
+                        // Read in the pubkey from key
+                        CPubKey pubkey(key.begin() + 1, key.end());
+                        if (!pubkey.IsFullyValid()) {
+                           throw std::ios_base::failure("Invalid pubkey");
+                        }
+
+                        // Read in key path
+                        std::vector<uint32_t> keypath;
+                        for (unsigned int i = 0; i < value_len; i += sizeof(uint32_t)) {
+                            uint32_t index;
+                            s >> index;
+                            keypath.push_back(index);
+                        }
+
+                        // Add to map
+                        hd_keypaths.emplace(pubkey, keypath);
+                    } else {
+                        // Read in the sighash type
+                        s >> input.sighash_type;
+                    }
+                    break;
+                // Number of inputs and input index
+                case PSBT_NUM_IN_VIN:
+                    if (in_globals) {
+                        num_ins = ReadCompactSize(s);
+                    } else {
+                        // Make sure that we are using inout indexes or this is the first input
```

Comment:
small typo: Make sure that we are using input indexes or this is the first input

Reply:
Fixed